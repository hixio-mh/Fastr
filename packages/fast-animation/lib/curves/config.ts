import { IBezierCurve } from "./index";

export const linear: IBezierCurve = [
    [0, 0],
    [1, 1]
];
export const easeOut: IBezierCurve = [
    [0, 0],
    [0.58, 1]
];
export const easeOutSmooth: IBezierCurve = [
    [0, 0.35],
    [0.15, 1]
];
export const easeIn: IBezierCurve = [
    [0.25, 0.1],
    [0.25, 1]
];
export const drillIn: IBezierCurve = [
    [0.17, 0.17],
    [0, 1]
];
export const backToApp: IBezierCurve = [
    [0.5, 0],
    [0.6, 1]
];
export const appToApp: IBezierCurve = [
    [0.5, 0],
    [1, 0.9]
];
export const fastIn: IBezierCurve = [
    [0.1, 0.9],
    [0.2, 1]
];
export const fastOut: IBezierCurve = [
    [0.9, 0.1],
    [1, 0.2]
];
export const fastInOut: IBezierCurve = [
    [0.8, 0],
    [0.2, 1]
];
export const exponential: IBezierCurve = [
    [0.1, 0.25],
    [0.75, 0.9]
];
export const fastInFortySevenPercent: IBezierCurve = [
    [0.11, 0.5],
    [0.24, 0.96]
];
export const exponentialReversed: IBezierCurve = [
    [0.25, 0.1],
    [0.9, 0.75]
];
export const navPane: IBezierCurve = [
    [0.1, 0.7],
    [0.1, 1]
];
