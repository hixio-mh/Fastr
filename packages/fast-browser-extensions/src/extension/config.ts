/**
 * Define configuration values
 */
const validDomains: string[] = ["*://*.fluentweb.com/*"];
const extensionId: string = "jhhigejkemaekdcempcigaapebobmimg";

/**
 * Define event names
 */
export const menuItemClickEvent: string = "FW_EXTENSION::MENU_ITEM_CLICK";

/**
 * Non-production environments
 */
if (process.env.NODE_ENV !== "production") {
    validDomains.push("<all_urls>");
}

/**
 * Assign exports
 */
export { validDomains, extensionId };
