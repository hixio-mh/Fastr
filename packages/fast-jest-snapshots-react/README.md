# FAST Jest snapshots React

`@microsoft/fast-jest-snapshots-react` is a small utility library for creating snapshot Jest tests for React components.
Given a single component, the `generateSnapshots` function will iterate over a set of prop data and create a snapshot test for each `prop` instance.

## Usage

To use, import the `ISnapshotTestSuite` interface and the `generateSnapshots` function from `@microsoft/fast-jest-snapshots-react`:

```ts
import {generateSnapshots, ISnapshotTestSuite} from "@microsoft/fast-jest-snapshots-react";
```

Then, create you test data snapshot suite, where every index of `data` is a valid prop object for your `component`:

```ts
// in a Jest test file
import {MyReactComponent, IMyReactComponentProps} from "my-react-component.tsx";
import {ISnapshotTestSuite, generateSnapshots} from "@microsoft/fast-jest-snapshots-react";

const testSuite: ISnapshotTestSuite<IMyReactComponentProps> = {
    name: 'test component',
    component: MyCustomReactComponentConstructor,
    data: [
        {
            id: "test-id",
            text: "hello world",
            // other component props
        }
    ]
};

// Generate snapshots using the prop and component data you defined
generateSnapshots(testSuite);
```

### Snapshot descriptions
Descriptions can optionally be applied to any snapshot for providing context to the test and the expectation of the test.
To do this, simply put the `prop` object into an array where the first index is the snapshot description and the second index is the `prop` object:

```ts
// in a Jest test file
import {MyReactComponent, IMyReactComponentProps} from "my-react-component.tsx";
import {ISnapshotTestSuite, generateSnapshots} from "@microsoft/fast-jest-snapshots-react";

const testSuite: ISnapshotTestSuite<IMyReactComponentProps> = {
    // ...
    data: [
        [
            "should render with an id attribute",
            {
                id: "test-id",
                text: "hello world",
                // other component props
            }

        ]
    ]
};
```
