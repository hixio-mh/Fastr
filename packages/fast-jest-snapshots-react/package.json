{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "sideEffects": false,
  "version": "2.3.0",
  "author": {
    "name": "Microsoft",
    "url": "https://discord.gg/FcSNfg4"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "build": "tsc -p ./tsconfig.json",
    "prepare": "npm run build",
    "start": "webpack-dev-server",
    "test": "npm run tslint",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'",
    "watch": "npm run build -- -w --preserveWatchOutput"
  },
  "jest": {
    "testURL": "http://localhost",
    "transform": {
      "^.+\\.tsx?$": "ts-jest",
      "^.+\\.jsx?$": "babel-jest"
    },
    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx"
    ]
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^2.2.0",
    "@types/jest": "^22.2.2",
    "@types/react": "^16.3.6",
    "jest": "^22.4.3",
    "react": "^16.5.2",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^3.0.1"
  },
  "dependencies": {
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
