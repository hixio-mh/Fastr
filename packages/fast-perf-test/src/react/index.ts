import React from "react";
import ReactDOM from "react-dom";
import { Checkbox, TextField } from "@microsoft/fast-components-react-msft";

const root = document.createElement("div");
document.body.appendChild(root);

function getQueryVariable(variable) {
    var query = window.location.search.substring(1);
    var vars = query.split("&");
    for (var i = 0; i < vars.length; i++) {
        var pair = vars[i].split("=");
        if (decodeURIComponent(pair[0]) == variable) {
            return decodeURIComponent(pair[1]);
        }
    }
}

const count = getQueryVariable("count");

function renderItems() {
    return React.createElement(
        React.Fragment,
        {},
        new Array(typeof count === "string" ? (count as any) * 1 : 2000)
            .fill(0)
            .map((item, index) => React.createElement(TextField))
    );
}

function render(): void {
    ReactDOM.render(React.createElement(renderItems), root);
}

render();
