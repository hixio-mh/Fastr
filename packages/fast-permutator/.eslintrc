{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
            "classes": true
        }
    },
    "rules": {
        "semi": 2
    }
}