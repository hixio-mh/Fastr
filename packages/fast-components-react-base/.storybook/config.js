import React from "react";
import { configure, addParameters, addDecorator } from "@storybook/react";
import { create, Global } from "@storybook/theming";
// Import from dist cause storybook throws up if we import the named import from the index
import { theme } from "@microsoft/fast-storybook-presets";

addParameters({
    options: {
        theme,
    },
});
