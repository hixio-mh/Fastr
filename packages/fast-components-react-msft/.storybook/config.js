import React from "react";
import { configure, addParameters, addDecorator } from "@storybook/react";
import { create, Global } from "@storybook/theming";
import { addReadme } from "storybook-readme";

// Import from dist cause storybook throws up if we import the named import from the index
import theme from "@microsoft/fast-storybook-presets/dist/theme";

addDecorator(addReadme);
addParameters({
    options: {
        theme,
    },
});
