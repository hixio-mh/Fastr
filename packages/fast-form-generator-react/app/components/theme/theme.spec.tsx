import * as React from "react";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import { ISnapshotTestSuite } from "@microsoft/fast-jest-snapshots-react";
import { getExample } from "@microsoft/fast-permutator";
import Form, { IFormProps } from "../../../src/form";
import config from "./theme.config";
import * as themeSchema from "./theme.schema.json";

const name: string = "theme";

const exampleData: IFormProps = {
    schema: themeSchema,
    data: getExample(themeSchema),
    /* tslint:disable-next-line */
    onChange: (data: any): void => {},
    componentMappingToPropertyNames: config
};

const examples: ISnapshotTestSuite<IFormProps> = {
    name,
    component: Form,
    data: [
        exampleData
    ]
};

describe(name, () => {
    generateSnapshots(examples);
});
