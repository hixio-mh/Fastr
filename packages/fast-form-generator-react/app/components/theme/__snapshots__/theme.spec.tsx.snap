// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme theme: 0 1`] = `
<div
  className={null}
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <div>
          <div>
            <div
              className="formItemTheme-0-1-11"
            >
              <label
                className="formItemTheme_label-0-1-12"
                htmlFor="theme"
              >
                Theme
              </label>
              <div
                className="formItemTheme_inputContainer-0-1-13"
              >
                <input
                  aria-label="theme light"
                  checked={true}
                  className="formItemTheme_input__light-0-1-14"
                  id="theme"
                  name="theme"
                  onChange={[Function]}
                  type="radio"
                  value="light"
                />
                <input
                  aria-label="theme dark"
                  checked={false}
                  className="formItemTheme_input__dark-0-1-15"
                  id="theme"
                  name="theme"
                  onChange={[Function]}
                  type="radio"
                  value="dark"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
